lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -4234,22 +4239,23 @@ class AdjointEmitter final : public SILInstructionVisitor<AdjointEmitter> {
             originalBuffer, getInvoker(),
             diag::autodiff_noderivative_stored_property);
         errorOccurred = true;
-        return (bufferMap[originalBuffer] = ValueWithCleanup());
+        return (bufferMap[{origBB, originalBuffer}] = ValueWithCleanup());
       }
     }
 
     // If the original buffer is a projection, return a corresponding projection
     // into the adjoint buffer.
-    if (auto adjProj = getAdjointProjection(originalBuffer)) {
+    if (auto adjProj = getAdjointProjection(origBB, originalBuffer)) {
       ValueWithCleanup projWithCleanup(
           adjProj, makeCleanup(adjProj, /*cleanup*/ nullptr));
-      return (bufferMap[originalBuffer] = projWithCleanup);
+      return (bufferMap[{origBB, originalBuffer}] = projWithCleanup);
     }
 
     // Set insertion point for local allocation builder: before the last local
     // allocation, or at the start of the adjoint function's entry if no local
     // allocations exist yet.
     localAllocBuilder.setInsertionPoint(
+        getAdjoint().getEntryBlock(),
```

Comment:
This line is crucial to prevent an `Assertion failed: (!NodePtr->isKnownSentinel())` crash.